generate_connectors.py

```diff
@@ -310,53 +306,39 @@ def _uuid(identifier):
-                lines.append(vertex.format(ff(spacing / 2 + dx), ff(y + dy)))
-                lines.append(vertex.format(ff(spacing / 2 + dx), ff(y - dy)))
-                lines.append(vertex.format(ff(spacing / 2 - dx), ff(y - dy)))
-                lines.append(vertex.format(ff(spacing / 2 - dx), ff(y + dy)))
-                lines.append(' )')
+                polygon = Polygon(uuid_decoration, Layer('sym_outlines'), Width(line_width), Fill(True), GrabArea(True))
+                polygon.add_vertex(Vertex(Position(spacing / 2 - dx, y + dy), Angle(0.0)))
+                polygon.add_vertex(Vertex(Position(spacing / 2 + dx, y + dy), Angle(0.0)))
+                polygon.add_vertex(Vertex(Position(spacing / 2 + dx, y - dy), Angle(0.0)))
+                polygon.add_vertex(Vertex(Position(spacing / 2 - dx, y - dy), Angle(0.0)))
+                polygon.add_vertex(Vertex(Position(spacing / 2 - dx, y + dy), Angle(0.0)))
+                symbol.add_polygon(polygon)
         elif kind == KIND_SOCKET:
             # Sockets: Small semicircle
             for j in range(1, i + 1):
                 y = get_y(j, i, spacing, True)
                 d = spacing / 4 * 0.75
                 w = line_width * 0.75
-                lines.append(' (polygon {} (layer sym_outlines)'.format(uuid_decoration))
-                lines.append('  (width {}) (fill false) (grab_area false)'.format(w))
-                lines.append('  (vertex (position {} {}) (angle 135.0))'.format(
-                    ff(spacing / 2 + d * 0.5 - d - w), ff(y - d)),
-                )
-                lines.append('  (vertex (position {} {}) (angle 0.0))'.format(
-                    ff(spacing / 2 + d * 0.5 - d - w), ff(y + d))
-                )
-                lines.append(' )')
+                polygon = Polygon(uuid_decoration, Layer('sym_outlines'), Width(w), Fill(False), GrabArea(False))
+                polygon.add_vertex(Vertex(Position(spacing / 2 + d * 0.5 - d - w, y - d), Angle(135.0)))
+                polygon.add_vertex(Vertex(Position(spacing / 2 + d * 0.5 - d - w, y + d), Angle(0.0)))
+                symbol.add_polygon(polygon)
 
         # Text
         y_max, y_min = get_rectangle_bounds(i, spacing, spacing, True)
-        lines.append(' (text {} (layer sym_names) (value "{{{{NAME}}}}")'.format(uuid_text_name))
-        lines.append('  (align center bottom) (height {}) (position 0.0 {}) (rotation 0.0)'.format(
-            ff(sym_text_height), ff(y_max),
-        ))
-        lines.append(' )')
-        lines.append(' (text {} (layer sym_values) (value "{{{{VALUE}}}}")'.format(uuid_text_value))
-        lines.append('  (align center top) (height {}) (position 0.0 {}) (rotation 0.0)'.format(
-            ff(sym_text_height), ff(y_min),
-        ))
-        lines.append(' )')
+        text = Text(uuid_text_name, Layer('sym_names'), Value('{{NAME}}'), Align('center bottom'), Height(sym_text_height), Position(0.0, y_max), Rotation(0.0))
+        symbol.add_text(text)
```

Comment:
We could add some dedicated helpers like `add_name` and `add_value` which use the conventions for align, position and rotation.